packages/docs/docs/core-api/user-state.md

```diff
@@ -101,12 +108,15 @@ state.stop(): void
 
 ## hasSignedUp
 
+Query the current database if the [Semaphore](https://semaphore.appliedzkp.org/) identity commitment is stored.
+
 ```ts
 state.hasSignedUp(attesterId?: bigint | string): Promise<boolean>
 ```
 
 ## latestTransitionedEpoch
 
+Query the current database where the latest [nullifier](../protocol/nullifiers.md) is emitted or signup in which epoch.
```

Comment:
i don't understand what you mean with the second part of this sentence
'... or signup in which epoch'

in `../protocol/nullifiers.md' where the link goes, it reads 'UniRep uses two different nullifiers:'
--but only one is listed

Reply:
`latestTransitionedEpoch` is the epoch which the user has a leaf in the state tree.
so there are two signals to determine whether the user has a leaf
1. a user state transition nullifier is emitted
2. the signup event is emitted

Is there a better description? 🙏🏻

Reply:
ok thanks for the explanation! i think it could read:
'Query the current database for a user's signup event or latest ust nullifier.'

Reply:
please check
[39b37f5](https://github.com/Unirep/Unirep/pull/513/commits/39b37f539acbe4b4553e741101eb5833c0929c5a)